library/VTKExtensions/Rendering/glsl/vtkF3DBitonicSortFunctions.glsl

```diff
@@ -0,0 +1,42 @@
+ivec2 disperse(uint t, uint h)
+{
+  uint q = (t / h) * 2 * h;
+
+  uint x = q + t % h;
+  uint y = q + t % h + h;
+
+  return ivec2(x, y);
+}
+
+ivec2 flip(uint t, uint h)
+{
+  uint q = (t / h) * 2 * h;
+
+  uint x = q + t % h;
+  uint y = q + 2 * h - (t % h) - 1;
+
+  return ivec2(x, y);
+}
+
+void swap_key(inout KeyType key1, inout KeyType key2)
+{
+  KeyType tmp = key1;
+  key1 = key2;
+  key2 = tmp;
+}
+
+void swap_value(inout ValueType value1, inout ValueType value2)
+{
+  ValueType tmp = value1;
+  value1 = value2;
+  value2 = tmp;
+}
+
+void compare_and_swap(ivec2 idx)
+{
+  if (key[idx.x] > key[idx.y] && idx.y < count)
```

Comment:
what is it `count` ?

Reply:
`count` is the number of gaussians here. The sort algorithm works only with power of two number of elements so we have trash after the gaussians. So I'm just checking if the second element is valid here (that's very standard in GPGPU)

Reply:
alright thanks for the explanation